activerecord/test/cases/migration/references_foreign_key_test.rb

```diff
@@ -63,7 +63,7 @@ class ReferencesForeignKeyInCreateTest < ActiveRecord::TestCase
                        fks.map { |fk| [fk.from_table, fk.to_table, fk.column] })
         end
 
-        if current_adapter?(:PostgreSQLAdapter)
+        if current_adapter?(:PostgreSQLAdapter) || current_adapter?(:SQLite3Adapter)
```

Comment:
How about using `supports_deferrable_constraints?` method to run tests?

```suggestion
        if ActiveRecord::Base.connection.supports_deferrable_constraints?
```